### PR TITLE
mgmt, keyvault refactor, avoid GET when list keys/secrets

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Separate List and Get. List only lists common part, key and secret value will be retrieved by a separate GET request.

- Added `enabled` method in `Secret`.
- Updated `getAttributes`, `getTags`, `isManaged` to `attributes`, `tags`, `managed` in `Key`.
- Updated `list` method in `Keys` and `Secrets`. It will no longer retrieve key and secret value. Key can be retrieved via `getJsonWebKey`. Secret value can be retrieved via `getValue`.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretImpl.java [hunk 1]
```diff
@@ -76,24 +98,20 @@ public boolean managed() {
         return vault
             .secretClient()
             .listPropertiesOfSecretVersions(name())
-            .flatMap(p -> {
-                if (p.isEnabled()) {
-                    return vault.secretClient().getSecret(p.getName(), p.getVersion());
-                } else {
-                    return Mono.just(new KeyVaultSecret(p.getName(), null).setProperties(p));
-                }
-            })
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretImpl.java:85]: @xccc-msft 
Now that I've removed the per-item GET request, and added the `enabled` method, I believe I can let customer handle this case. Customer should be able to check `enabled` and decide whether to do a `getValue` or not.

Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretImpl.java:85]: The change looks good. One question is for the secret version. When the customer want to get value for enabled secret, I think `.getSecret(name, version)` should have version param passed too. Not sure if I missed any part, but it seems it alwarys pass NULL as per your changes.

File: sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretImpl.java [hunk 2]
```diff
@@ -76,24 +98,20 @@ public boolean managed() {
...
             .map(this::wrapModel);
     }
 
     @Override
-    protected Mono<KeyVaultSecret> getInnerAsync() {
-        return vault.secretClient().getSecret(name(), null);
+    protected Mono<SecretProperties> getInnerAsync() {
+        return vault.secretClient().getSecret(name(), innerModel().getVersion()).map(secret -> {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretImpl.java:106]: Get the `version` param fixed.

**Review by xccc-msft — APPROVED** [review]
LGTM

**Comment by weidongxu-microsoft** [comment]
`getSecret` will have exception when disabled. This is expected.

```
com.azure.core.exception.ResourceModifiedException: Status code 403, "{"error":{"code":"Forbidden","message":"Operation get is not allowed on a disabled secret.","innererror":{"code":"SecretDisabled"}}}"
```

**Comment by yungezz** [comment]
> `getSecret` will throw exception when disabled. This is expected.
> 
> ```
> com.azure.core.exception.ResourceModifiedException: Status code 403, "{"error":{"code":"Forbidden","message":"Operation get is not allowed on a disabled secret.","innererror":{"code":"SecretDisabled"}}}"
> ```

user should pass in `enabled` true when calling getSecret?

**Comment by weidongxu-microsoft** [comment]
> user should pass in `enabled` true when calling getSecret?

Usually secret is enabled.

If it is not enabled, service will not allow getting its value, until it is enabled again.

I think I will add `enable` and `disable` on `Secrets` interface to provide the enable/disable operation.

**Review by yungezz — APPROVED** [review]

**Pull request merged**